lib/gl/shader.cpp

```diff
@@ -181,16 +182,28 @@ std::string ShaderProgram::formatShader(const std::string& inShader,
    {
       formatted = "#define USE_ALPHA\n" + formatted;
    }
-   // add the header
-   formatted = std::regex_replace(GLSL_HEADER, std::regex("GLSL_VER"),
-                                  std::to_string(glsl_version)) + formatted;
-#ifdef __EMSCRIPTEN__
-   // special prepend for WebGL 2 shaders
-   if (glsl_version == 300)
+
+   if (glsl_is_es)
    {
-      formatted = "#version 300 es\n" + formatted;
+      // Add precision specifier - required for WebGL
+      formatted = "precision mediump float;\n" + formatted;
```

Comment:
What does `precision mediump float` mean?

Reply:
They specify the precision of declared `floats` in the shader. `precision highp float` specifies a minimum 16-bit mantissa, `mediump` has 10 bits, and `lowp` has 8 bits.

The reason we need this for WebGL is because OpenGL ES doesn't specify a default precision for fragment shader variables. Also, support for `precision highp` in the fragment shader isn't guaranteed on all platforms, especially mobile devices.

Although, we should probably make this addition conditional on `if (shaderType == GL_FRAGMENT_SHADER)`, since we do want high-precision FP operations in the vertex shader.